package.yml: Add new macros (#536)
getsolus/help-center-docs: docs/packaging/package.yml.md

### Actionable macros

| Macro                         | Description                                                                                                                 |
| ----------------------------- | --------------------------------------------------------------------------------------------------------------------------- |
| **%autogen**                  | Runs autogen with our `%CONFOPTS%` to create a configure script then proceeds to run `%configure`.                          |
| **%cmake**                    | Configures a cmake project with the distribution specific options, such as prefix and release type.                         |
| **%cmake_ninja**              | Configures a cmake project with ninja so it can be used with `%ninja_build`, `%ninja_install` and `%ninja_check` macros.    |
| **%configure**                | Runs `./configure` with our `%CONFOPTS%` variable macro.                                                                    |

> | **%configure_no_runstatedir** | Use if `%configure` returns `configure: error: unrecognized option: '--runstatedir=/run'`                                   |

Reviewer: This entry is a bit different from the others in that it doesn't describe what it does.

Maybe add something like "Like `%configure` but without the `--runstatedir` argument. Use if [...]"

Plus missing period at the end ^^

| **%make**                     | Runs the `make` command with the job count specified in `eopkg.conf`. ([More info](advanced-config/eopkg-configuration.md)) |
| **%make_install**             | Performs a `make install`, using the `DESTDIR` variant. Should work for the vast majority of packages.                      |
| **%patch**                    | Sane patch macro to run in batch mode and not contaminate source tree on failure.                                           |
| **%apply_patches**            | Applies all patches listed in the `series` file in `./files` folder.                                                        |
| **%reconfigure**              | Updates build scripts such as `./configure` and proceeds to run `%configure`.                                               |

> | **%symlink_check**            | Installed symlinks are checked; if a broken symlink is found the build will fail. Must run after install macros             |

Reviewer: ```suggestion
| **%symlink_check**            | Checks for broken symlinks in the install directory and aborts the build if any are found. Must run after install macros.             |
```

I know I approved the previous version on Matrix, but now that I'm sitting down at my computer maybe something like this would be a bit better? (also closer to the phrasing for other entries in the table). Plus period at the end.
